samza-core/src/main/java/org/apache/samza/zk/ZkUtils.java

```diff
@@ -123,17 +154,60 @@ public synchronized String getEphemeralPath() {
   }
 
   /**
-   * Method is used to get the <i>sorted</i> list of currently active/registered processors
+   * Method is used to get the <i>sorted</i> list of currently active/registered processors (znodes)
    *
    * @return List of absolute ZK node paths
    */
   public List<String> getSortedActiveProcessors() {
-    List<String> children = zkClient.getChildren(keyBuilder.getProcessorsPath());
-    if (children.size() > 0) {
-      Collections.sort(children);
-      LOG.info("Found these children - " + children);
+    List<String> znodeIds = zkClient.getChildren(keyBuilder.getProcessorsPath());
+    if (znodeIds.size() > 0) {
+      Collections.sort(znodeIds);
+      LOG.info("Found these children - " + znodeIds);
+    }
+    return znodeIds;
+  }
+
+  /**
+   * Method is used to read processors PIDs
+   * @param fullPath absolute path to the znode
+   * @return absolute znode path
+   */
+  public String getProcessorsPID(String fullPath) {
+    String data = zkClient.<String>readData(fullPath, true);
+    if (data == null) {
+      throw new SamzaException(String.format("Cannot read ZK node:", fullPath));
+    }
+    return new ProcessorData(data).getProcessorId();
+  }
+
+  /**
+   * Method is used to get the <i>sorted</i> list of currently active/registered processor ids
+   * @return List of processorIds
+   */
+  public List<String> getSortedActiveProcessorsPIDs() {
```

Comment:
Please fix the "PIDs" in the method name.

Why is this overload method needed?? When should the caller use one Vs the other? Please clarify it in the docs or change this pattern.